internal/server/proto/server.proto

```diff
@@ -767,7 +767,7 @@ message Ref {
 
   // DeclaredResource references a declared resource.
   message DeclaredResource {
-    string id = 1;
+    string name = 1;
```

Comment:
As it turns out, resource manager resource names are supposed to be unique, so now we're using that `name` instead of generating a unique id.

This brings this proto in line with the corresponding sdk proto.